components/status_dropdown/status_dropdown.tsx

```diff
@@ -24,76 +23,81 @@ import StatusOfflineIcon from 'components/widgets/icons/status_offline_icon';
 import OverlayTrigger from 'components/overlay_trigger';
 import CustomStatusText from 'components/custom_status/custom_status_text';
 
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import {UserCustomStatus, UserStatus} from 'mattermost-redux/types/users';
+
 import './status_dropdown.scss';
 
-export default class StatusDropdown extends React.PureComponent {
-    static propTypes = {
-        style: PropTypes.object,
-        status: PropTypes.string,
-        userId: PropTypes.string.isRequired,
-        profilePicture: PropTypes.string,
-        autoResetPref: PropTypes.string,
-        actions: PropTypes.shape({
-            openModal: PropTypes.func.isRequired,
-            setStatus: PropTypes.func.isRequired,
-            unsetCustomStatus: PropTypes.func.isRequired,
-            setStatusDropdown: PropTypes.func.isRequired,
-        }).isRequired,
-        customStatus: PropTypes.object,
-        isCustomStatusEnabled: PropTypes.bool.isRequired,
-        isStatusDropdownOpen: PropTypes.bool.isRequired,
-        showCustomStatusPulsatingDot: PropTypes.bool.isRequired,
-    }
+type Props = {
+    status?: string;
+    userId: string;
+    profilePicture: string;
+    autoResetPref?: string;
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
+        setStatus: (status: UserStatus) => ActionFunc;
+        unsetCustomStatus: () => ActionFunc;
+        setStatusDropdown: (open: boolean) => void;
+    };
+    customStatus: UserCustomStatus | Record<string, any>;
```

Comment:
How come we need to union with `Record<string, any>` here?

Reply:
Hi @devinbinnie, it is because the defaultProps of customStatus in components/status_dropdown/status_dropdown.tsx is empty object.

<img width="1168" alt="Screen Shot 2021-05-03 at 22 35 49" src="https://user-images.githubusercontent.com/28821618/116898090-2c991580-ac60-11eb-800d-a6c1bdc86952.png">

Reply:
Can we try casting it as `UserCustomStatus`?

Reply:
Done @devinbinnie